### PR TITLE
DEV-6390 change Agency v2 URLs

### PR DESCRIPTION
**Pull request by ebdabbs**
**High level description:**

Use the agency name slug (e.g. `department-of-the-treasury`) instead of the toptier code (e.g. `020`) in (dark-released) Agency Profile v2 URLs

**Technical details:**
- Implements a custom hook, `useAgencySlugs`, which returns a mapping of `agency_slug` to `toptier_code` if it exists in Redux. If it does not exist in Redux yet, it makes an API request to the endpoint `v2/references/toptier_agencies`
- Handles the loading state for each section of the page in `AgencySection.jsx` instead of per-component
- Fixes a few console errors about unique keys
- In the containers, get `toptierCode` from the `agencyV2.overview` object Redux (instead of passing it down as a prop or getting it from the URL)
- Bonus! Removes the download button from the page header to match the latest mocks, since I was already in there editing the share button logic

**JIRA Ticket:**
[DEV-6390](https://federal-spending-transparency.atlassian.net/browse/DEV-6390)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Provided instructions for testing in JIRA
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [x] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)

Reviewer(s):
- [x] Code review complete

### CODE REVIEW

File: src/js/containers/agencyV2/AgencyContainerV2.jsx [hunk 1]
```diff
@@ -15,28 +15,33 @@ import { setAgencyOverview } from 'redux/actions/agencyV2/agencyV2Actions';
...
     const [isError, setError] = useState(false);
     const [errorMessage, setErrorMessage] = useState('');
+    const [redirect, setRedirect] = useState(false);
     const request = useRef(null);
     const dispatch = useDispatch();
+    const [toptierCode, setToptierCode] = useState(agencySlugs[agencySlug]);
 
     useEffect(() => {
-        if (selectedFy) {
+        if (selectedFy && toptierCode) {
```
Comment (afnazu) [inline src/js/containers/agencyV2/AgencyContainerV2.jsx:36]: Question for my own knowledge! why do we now need to include toptierCode in this useEffect when previously we just needed selectedFY to make the request?

Comment (ebdabbs) [reply src/js/containers/agencyV2/AgencyContainerV2.jsx:36]: Good question! Before, we were getting toptier code (agency ID) from the URL parameter. So as soon as you landed on the page, it was available in the URL. Now, we have to wait for the API request made by `useAgencySlugs` to complete before the toptierCode is available.

File: src/js/containers/agencyV2/WithAgencySlugs.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        };
+    }, [agencySlugs]);
+
+    return [agencySlugs, loading, error];
+};
+
+const withAgencySlugs = (WrappedComponent) => (props) => {
+    const [agencySlugs, loading, error] = useAgencySlugs();
+    return (
+        <WrappedComponent
```
Comment (afnazu) [inline src/js/containers/agencyV2/WithAgencySlugs.jsx:59]: Nice I haven't seen higher order components before great use case 👌

**Review by afnazu — APPROVED** [review]
Looks great- no issues on my end! thanks for resolving those unique key console errors 🎉

**Pull request merged**